src/IonEventStream.ts

```diff
@@ -149,46 +157,67 @@ export class IonEventStream {
         return true;
     }
 
-    private generateStream(): void {
-        let tid: IonType | null = this.reader.next();
-        if (tid === IonTypes.SYMBOL && this.reader.stringValue() === "ion_event_stream") {
-            this.marshalStream();
-            return;
-        }
-        let currentContainer: IonEvent[] = [];
-        let currentContainerIndex: number[] = [];
-        while (true) {
-            if (this.reader.isNull()) {
-                this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid!, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), true, this.reader.value()));
-            } else {
-                switch (tid) {
-                    case IonTypes.LIST:
-                    case IonTypes.SEXP:
-                    case IonTypes.STRUCT: {
-                        let containerEvent = this.eventFactory.makeEvent(IonEventType.CONTAINER_START, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, null);
-                        this.eventStream.push(containerEvent);
-                        currentContainer.push(containerEvent);
-                        currentContainerIndex.push(this.eventStream.length);
-                        this.reader.stepIn();
-                        break;
-                    }
-                    case null: {
-                        if (this.reader.depth() === 0) {
-                            this.eventStream.push(this.eventFactory.makeEvent(IonEventType.STREAM_END, IonTypes.NULL, null, this.reader.depth(), [], false, undefined));
-                            return;
-                        } else {
-                            this.reader.stepOut();
-                            this.endContainer(currentContainer.pop()!, currentContainerIndex.pop()!);
+    private generateStream(path: string, args: IonCliCommonArgs): void {
+        try{
+            let tid: IonType | null = this.reader.next();
+            if (tid === IonTypes.SYMBOL && this.reader.stringValue() === "$ion_event_stream") {
+                this.marshalStream();
+                return;
+            }
+            if (tid === IonTypes.SEXP || IonTypes.LIST && this.reader.annotations()[0] === "embedded_documents") {
+                this.isEmbedded = true;
+            }
+            let currentContainer: IonEvent[] = [];
+            let currentContainerIndex: number[] = [];
+            while (true) {
+                if (this.reader.isNull()) {
+                    this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid!, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), true, this.reader.value()));
```

Comment:
Some of these lines are very wide; could you see if your IDE has an auto-formatter? 

Note: if you autoformat the code and it changes a huge number of lines, don't worry about it. We'll finish this PR and then we can open a separate PR to do sweeping, formatting-only changes. Once that's done we can figure out how to add formatting requirements to our build pipeline.